src/Deployment/FtpServer.php

```diff
@@ -278,10 +277,25 @@ private function protect(callable $func, $args = [])
 				$message = $m[1];
 			}
 			throw new FtpException($message);
-		});
-		$res = call_user_func_array($func, $args);
+		};
+
+		set_error_handler($ftpExceptionErrorHandler);
```

Comment:
It can be done this way:

```php
		set_error_handler(function ($severity, $message) {
			if (ini_get('html_errors')) {
				$message = html_entity_decode(strip_tags($message));
			}
			if (preg_match('#^\w+\(\):\s*(.+)#', $message, $m)) {
				$message = $m[1];
			}
			throw new FtpException($message);
		});

		try {
			$res = call_user_func_array($func, $args);
			restore_error_handler();
		} catch (\Exception $e) {
			restore_error_handler();
			throw $e;
		}
		return $res;
```